it/controllers/benefits/income/IncomeTaxBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,546 @@
+                headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            s"has an BAD REQUEST($BAD_REQUEST) status" in {
+              result.status shouldBe BAD_REQUEST
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+            h1Check(user.specificExpectedResults.get.expectedHeading)
+            textOnPageCheck(optionalParagraphText(amount), paragraphSelector(2))
+            textOnPageCheck(enterTotalText, paragraphSelector(3))
+            hintTextCheck(expectedHintText, hintTextSelector)
+            textOnPageCheck(currencyPrefix, currencyPrefixSelector)
+            inputFieldCheck(amountFieldName, amountFieldSelector)
+            inputFieldValueCheck(overMaximumAmount, amountFieldSelector)
+            buttonCheck(continueButtonText, continueButtonSelector)
+            errorSummaryCheck(user.specificExpectedResults.get.expectedErrorOverMaximum, expectedErrorHref)
+            errorAboveElementCheck(user.specificExpectedResults.get.expectedErrorOverMaximum, Some(amountFieldName))
+            formPostLinkCheck(formLink, formSelector)
+
+            welshToggleCheck(user.isWelsh)
+          }
+        }
+      }
+    }
+
+    "Update the incomeTaxPaidByDirector value when a user submits a valid form and has prior benefits, redirects to CYA" which {
+    val newAmount: BigDecimal = 123.45
+    val form: Map[String, String] = Map(AmountForm.amount -> newAmount.toString())
+
+    implicit lazy val result: WSResponse = {
+      authoriseAgentOrIndividual(isAgent = false)
+      dropEmploymentDB()
+      userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+      insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel(Some(benefits(fullIncomeTaxOrIncurredCostsModel)))), userRequest)
+      urlPost(pageUrl(taxYearEOY), follow = false, body = form, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "redirects to check your benefits page" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+
+      "updates incomeTaxPaidByDirector value to the new amount" in {
+        lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+        cyamodel.employment.employmentBenefits.flatMap(_.incomeTaxAndCostsModel.flatMap(_.incomeTaxPaidByDirectorQuestion)) shouldBe Some(true)
+        cyamodel.employment.employmentBenefits.flatMap(_.incomeTaxAndCostsModel.flatMap(_.incomeTaxPaidByDirector)) shouldBe Some(newAmount)
+      }
+    }
+
+    "Update the incomeTaxPaidByDirector value when a user submits a valid form and has no prior benefits, redirects to incurred costs" which {
```

Comment:
can you add the updated value check to this test like you did for the above test